From eef37f2d89e72954645f4f3583227df74c841a84 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Mateusz=20Miku=C5=82a?= <mati865@gmail.com>
Date: Sun, 18 Sep 2022 13:17:16 +0200
Subject: [PATCH] List memrchr as unavailable on Windows

Otherwise LLVM will optimize strrchr into memrchr causing linker error
---
 lib/Analysis/TargetLibraryInfo.cpp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/lib/Analysis/TargetLibraryInfo.cpp b/lib/Analysis/TargetLibraryInfo.cpp
index 75b57e75fdd0..40f0cf39d7d6 100644
--- a/lib/Analysis/TargetLibraryInfo.cpp
+++ b/lib/Analysis/TargetLibraryInfo.cpp
@@ -439,6 +439,7 @@ static void initialize(TargetLibraryInfoImpl &TLI, const Triple &T,
     TLI.setUnavailable(LibFunc_htons);
     TLI.setUnavailable(LibFunc_lchown);
     TLI.setUnavailable(LibFunc_lstat);
+    TLI.setUnavailable(LibFunc_memrchr);
     TLI.setUnavailable(LibFunc_ntohl);
     TLI.setUnavailable(LibFunc_ntohs);
     TLI.setUnavailable(LibFunc_pread);
-- 
2.37.3.windows.1

